src/alchemy/alchemy.service.ts

```diff
@@ -156,34 +156,42 @@ export class AlchemyService {
         }
     }
 
-    async getTokenHistoricPrice(symbol: string, beginDate: Date, endDate: Date, interval: Number, tokenAddress: string, network: string): Promise<any> {
-        const formattedBeginDate = beginDate.toISOString();
-        const formattedEndDate = endDate.toISOString();
+    async getTokenHistoricPrices(tokenAddress: string, startTimes: string[],interval: string): Promise<{ price: number; date: string }[]> {
         const url = `https://api.g.alchemy.com/prices/v1/${this.apiKey}/tokens/historical`;
-        console.log(tokenAddress, network);
-        try {
+        const results: { price: number; date: string }[] = [];
+        
+        for (const startTime of startTimes) {
+            const beginDate = new Date(startTime);
+            const formattedBeginDate = beginDate.toISOString();
+            const formattedEndDate = new Date().toISOString();
+
             const data = {
                 startTime: formattedBeginDate,
                 endTime: formattedEndDate,
                 interval: interval,
                 address: tokenAddress,
-                network: network
```

Comment:
![image](https://github.com/user-attachments/assets/78aa867b-2d7e-4d26-a879-b7700ee67eec)
i think even if it's useless for now it would be great to keep to keep the network open to something else

Reply:
The price is always the same depending of the network so not an issue